homeassistant/components/adaptive_lighting/switch.py

```diff
@@ -0,0 +1,1060 @@
+            min_color_temp=data[CONF_MIN_COLOR_TEMP],
+            sleep_brightness=data[CONF_SLEEP_BRIGHTNESS],
+            sleep_color_temp=data[CONF_SLEEP_COLOR_TEMP],
+            sunrise_offset=data[CONF_SUNRISE_OFFSET],
+            sunrise_time=data[CONF_SUNRISE_TIME],
+            sunset_offset=data[CONF_SUNSET_OFFSET],
+            sunset_time=data[CONF_SUNSET_TIME],
+            time_zone=self.hass.config.time_zone,
+        )
+
+        # Set other attributes
+        self._icon = ICON
+        self._state = None
+
+        # Tracks 'off' → 'on' state changes
+        self._on_to_off_event: Dict[str, Event] = {}
+        # Tracks 'on' → 'off' state changes
+        self._off_to_on_event: Dict[str, Event] = {}
+        # Locks that prevent light adjusting when waiting for a light to 'turn_off'
+        self._locks: Dict[str, asyncio.Lock] = {}
+        # To identify that this integration made a change
+        self.__context = Context()  # self._context will be overwritten
+
+        # Set in self._update_attrs_and_maybe_adapt_lights
+        self._settings: Dict[str, Any] = {}
+
+        # Set and unset tracker in async_turn_on and async_turn_off
+        self.remove_listeners = []
+        _LOGGER.debug(
+            "%s: Setting up with '%s',"
+            " config_entry.data: '%s',"
+            " config_entry.options: '%s', converted to '%s'.",
+            self._name,
+            self._lights,
+            config_entry.data,
+            config_entry.options,
+            data,
+        )
+
+    @property
+    def name(self):
+        """Return the name of the device if any."""
+        return f"Adaptive Lighting: {self._name}"
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of entity."""
+        return self._name
+
+    @property
+    def is_on(self) -> Optional[bool]:
+        """Return true if adaptive lighting is on."""
+        return self._state
+
+    async def async_added_to_hass(self) -> None:
+        """Call when entity about to be added to hass."""
+        if self.hass.is_running:
+            await self._setup_listeners()
+        else:
+            self.hass.bus.async_listen_once(
+                EVENT_HOMEASSISTANT_START, self._setup_listeners
+            )
+        last_state = await self.async_get_last_state()
+        is_new_entry = last_state is None  # newly added to HA
+        if is_new_entry or last_state.state == STATE_ON:
+            await self.async_turn_on(adapt_lights=not self._only_once)
+        else:
+            self._state = False
+            assert not self.remove_listeners
+
+    async def async_will_remove_from_hass(self):
+        """Remove the listeners upon removing the component."""
+        self._remove_listeners()
+
+    def _expand_light_groups(self) -> None:
+        all_lights = _expand_light_groups(self.hass, self._lights)
+        self.turn_on_off_listener.lights.update(all_lights)
+        self._lights = list(all_lights)
```

Comment:
@balloob, @frenck, this function is called after `EVENT_HOMEASSISTANT_START` and in 0.116.0b4 the lights are never expanded. I think this is a new bug in Home Assistant.

(`_expand_light_groups` turns a light group into the lights in the group.)

Reply:
@frenk helped me on Discord. I needed to listen to `EVENT_HOMEASSISTANT_STARTED` instead of `EVENT_HOMEASSISTANT_START`.

I guess some recent change changed the order in which things happen.

Fixed in d219b6d00b6724ec30627edf4c20f105272d6127.